backend/sozisel/lib/sozisel_web/schema/resolvers/presenter_resolvers.ex

```diff
@@ -0,0 +1,93 @@
+defmodule SoziselWeb.Schema.Resolvers.PresenterResolvers do
+  alias SoziselWeb.Schema.Helpers
+  alias SoziselWeb.Schema.Subscriptions.Topics
+
+  alias Sozisel.Model.{
+    Sessions,
+    Events,
+    LaunchedEvents
+  }
+
+  alias Events.Event
+  alias LaunchedEvents.LaunchedEvent
+  alias Sessions.Session
+
+  def prepare_data_for_participants(
+        %Event{} = event,
+        %LaunchedEvent{} = launched_event
+      ) do
+    event_data = event.event_data
+
+    participant_event_data = Events.marshal_participant_event_data(event_data)
+
+    %{
+      id: launched_event.id,
+      name: event.name,
+      event_data: participant_event_data
+    }
+  end
+
+  def send_launched_event_to_participants(
+        _parent,
+        %{
+          event_id: event_id,
+          session_id: session_id,
+          broadcast: broadcast,
```

Comment:
dont' pattern match the broadcast and target participants here as if target_participants gets nil it probably won't occur here and you will fail to pattern match on the whole function call
```elixir
%{
  event_id: event_id,
  session_id: session_d
} = attrs
```
and pass that `attrs` to the publish_event